### DISCUSSION TITLE
Switching from Live to VOD or VOD to Live

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#6698] [Q&A]
We will use two applications to switch from VOD to Live or Live to VOD. Let's go step by step.

**Step-1:**

In this example, we will use the `LiveApp` app for live streaming and the `live` app for VOD. These two applications are installed by default on the Ant Media Server. You have the option of creating your own applications.

**Step-2:**

First, go to the `LiveApp` application and create one live stream with streamId `test` for example.

![image](https://github.com/user-attachments/assets/154e19c5-c85d-4e28-8114-377dd9dcd80b)

**Step-3:**

Now go to the `LiveApp` application's **Settings --> Advanced** settings and set the below property:

```json
  "hlsPlayListType": "event",
```

After making the changes, save the settings.

**Step-4:**

Switch to the `live` application's VOD section and upload the VOD that you want to use. Check out [this document](https://antmedia.io/docs/guides/publish-live-stream/playlist/creating-playlist/#uploading-the-vod-files) for VOD upload.

**Step-5:**

After uploading the VOD, switch to live application **Settings --> Advanced** settings and make the below changes to the properties:

```json
  "deleteHLSFilesOnEnded": false,
  "hlsflags": "delete_segments+append_list+omit_endlist",
```

After making the changes, save the settings.

**Step-6:**

Open the Live Streams section of the `live` app and create a playlist with the streamId `playlist`, for example.

![image](https://github.com/user-attachments/assets/ff117903-76c9-43a7-9b42-c367e6eb56c1)

**Step-7:**

After creating the playlist stream, edit the playlist stream and first add the VOD that was uploaded.

![image](https://github.com/user-attachments/assets/090c8363-a3a5-4914-bf75-4ac09485d9fe)

**NOTE: The URL should be correct and reachable; otherwise, it will not be added.**

- Add the URL of the live stream that publishes to the LiveApp application with streamId test.

![image](https://github.com/user-attachments/assets/e8b6ea10-7e50-4190-adca-d544bca8f35d)

**NOTE: The URL should be correct and reachable; otherwise, it will not be added.**

- Now, after adding both the VOD URL and Live stream URL, save and update the playlist

**Step-8:**

Now, start the playlist and play with the below sample URL

**https://domain:5443/live/play.html?name=playlist&playOrder=hls**

When the playlist will be running, it will show the index that is currently running.

![image](https://github.com/user-attachments/assets/f0520cff-b864-493b-a1bd-fbba5da789bf)

Suppose the `Playing Index` is 0, which means the VOD is running and once the VOD time is over, if the actual live stream is live in the `LiveApp` application, then the playlist will switch to the live. In case the actual stream is not live, then the VOD will keep playing on loop.

Please feel free to comment if any points are unclear. We will do our best to include the demonstration video as soon as possible.

**Comment by klasadoroty** [comment]
hi,
I just tested it. Good job. It works without a problem.
I will add:
- when the playlist switches to streamlive it will be there all the time until the broadcast is turned off.
- when the broadcaster ends the 'live' operation it automatically returns to 'vod'.
- when the broadcaster starts streamlive again .. the live broadcast will appear.

Note: the vod file should be a maximum of 15s /at your discretion of course/, it works in a loop /check loop/.
I have not checked, but.. you do vod1 / live / vod2 / live / vod3 / live ... etc. - then you have a playlist of vod materials, but after starting the broadcast .. live will appear.

Note: before 'live' appears the server will finish the vod file.

**Reply by yashtandon113** [reply]
Hi @klasadoroty 

It is good to know you tried it and it works for you.

Also, thank you for adding the notes; I will include them in the official switching document to help users.

**Reply by klasadoroty** [reply]
Hi. One more thing: how do I make the playlist start when AMS starts? Automatically. If possible: please explain

**Reply by yashtandon113** [reply]
@klasadoroty 

After a server restart, the playlist does not start automatically. However, you can start the playlist using the Rest API. Check out this **[API call](https://antmedia.io/rest/#/default/startStreamSourceV2)**.

I hope it helps.

**Reply by klasadoroty** [reply]
Hi, didn't help... :)  
I know how to start. -the playlist does not start automatically- this is a problem.
